packages/cli/package.json

```diff
@@ -29,6 +29,7 @@
   },
   "homepage": "https://github.com/aws-amplify/amplify-backend#readme",
   "dependencies": {
+    "@aws-amplify/backend-deployer": "^0.4.5",
```

Comment:
This is a good catch.

I wonder why this rule didn't catch this earlier https://github.com/aws-amplify/amplify-backend/blob/3eb59ff4c709a9fe0fb02638312bb6c0fad01be8/.eslintrc.cjs#L122 .

Otherwise I wonder if we should declare `backend-deployer` as `peerDependency` given that in customer's app it's supposed to be delivered by `backend` package. @edwardfoyle what do you think?

Reply:
I put it under "dependencies" because I saw other `aws-amplify/*` packages are all in this category...

Reply:
It should be under dependencies. We should figure out why the lint rule didn't catch it.

Reply:
I've run the rule in debugger and it seems that rule silently skips it because it can't load the package.
See screenshots.
![image](https://github.com/aws-amplify/amplify-backend/assets/5849952/698e45cb-e484-4932-858e-cc0857da7d83)
![image](https://github.com/aws-amplify/amplify-backend/assets/5849952/c341dc58-15e3-463d-a9cf-7447cb6aae4b)

We should log GH issue to figure this out.

Reply:
Issue is created https://github.com/aws-amplify/amplify-backend/issues/908